tools/src/main/java/org/apache/kafka/tools/streams/StreamsGroupCommand.java

```diff
@@ -330,13 +403,433 @@ Map<TopicPartition, OffsetsInfo> getOffsets(StreamsGroupDescription description)
+                    .collect(Collectors.toMap(Function.identity(), topicPartition -> {
+                        OffsetAndMetadata committedOffset = currentCommittedOffsets.get(topicPartition);
+
+                        if (committedOffset == null) {
+                            throw new IllegalStateException("Expected a valid current offset for topic partition: " + topicPartition);
+                        }
+
+                        return new OffsetAndMetadata(committedOffset.offset());
+                    }));
+
+                Map<TopicPartition, OffsetAndMetadata> preparedOffsetsForPartitionsWithoutCommittedOffset = getLogEndOffsets(partitionsToResetWithoutCommittedOffset)
+                    .entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, e -> {
+                        if (!(e.getValue() instanceof LogOffset)) {
+                            CommandLineUtils.printUsageAndExit(opts.parser, "Error getting ending offset of topic partition: " + e.getKey());
+                        }
+
+                        return new OffsetAndMetadata(((LogOffset) e.getValue()).value);
+                    }));
+
+                preparedOffsetsForPartitionsWithCommittedOffset.putAll(preparedOffsetsForPartitionsWithoutCommittedOffset);
+
+                return preparedOffsetsForPartitionsWithCommittedOffset;
+            }
+
+            CommandLineUtils.printUsageAndExit(opts.parser, String.format("Option '%s' requires one of the following scenarios: %s", opts.resetOffsetsOpt, opts.allResetOffsetScenarioOpts));
+            return null;
+        }
+
+        Optional<Map<String, Map<TopicPartition, OffsetAndMetadata>>> resetPlanFromFile() {
+            if (opts.options.has(opts.resetFromFileOpt)) {
+                try {
+                    String resetPlanPath = opts.options.valueOf(opts.resetFromFileOpt);
+                    String resetPlanCsv = Utils.readFileAsString(resetPlanPath);
+                    Map<String, Map<TopicPartition, OffsetAndMetadata>> resetPlan = parseResetPlan(resetPlanCsv);
+                    return Optional.of(resetPlan);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            } else return Optional.empty();
+        }
+
+        private Map<String, Map<TopicPartition, OffsetAndMetadata>> parseResetPlan(String resetPlanCsv) {
+            ObjectReader csvReader = CsvUtils.readerFor(CsvUtils.CsvRecordNoGroup.class);
+            String[] lines = resetPlanCsv.split("\n");
+            boolean isSingleGroupQuery = opts.options.valuesOf(opts.groupOpt).size() == 1;
+            boolean isOldCsvFormat = false;
+            try {
+                if (lines.length > 0) {
+                    csvReader.readValue(lines[0], CsvUtils.CsvRecordNoGroup.class);
+                    isOldCsvFormat = true;
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+                // Ignore.
```

Comment:
Don't we want to fail here somehow?

Reply:
isn't it failing? Should we throw the exception you mean?

Reply:
Yes. It seems we print the stack trace and just continue.